paymentapi/src/main/java/fi/hel/verkkokauppa/payment/api/PaymentAdminController.java

```diff
@@ -124,22 +124,26 @@ public ResponseEntity<Void> orderCreatedEventCallbackPaytrail(@RequestBody Order
                     PaymentCardInfoDto card = new PaymentCardInfoDto(message.getCardToken(), message.getCardExpYear(), message.getCardExpMonth(), message.getCardLastFourDigits());
                     try {
                         PaytrailPaymentContext context = paymentPaytrailService.buildPaytrailContext(message.getNamespace(), message.getMerchantId());
+                        PaymentReturnDto paymentReturnDto = new PaymentReturnDto(true, false, false, false);
+                        String transactionId = null;
                         try {
-
                             PaytrailPaymentMitChargeSuccessResponse mitCharge = paymentPaytrailService.createMitCharge(context, payment.getPaymentId(), message);
-                            onlinePaymentService.setPaytrailTransactionId(payment.getPaymentId(), mitCharge.getTransactionId());
-                            onlinePaymentService.updatePaymentStatus(payment.getPaymentId(), new PaymentReturnDto(true, true, false, false), card);
+                            paymentReturnDto = new PaymentReturnDto(true, true, false, false);
+                            transactionId = mitCharge.getTransactionId();
                         } catch (Exception e) {
                             log.debug("subscription renewal payment failed for orderId: " + payment.getOrderId(), e);
-                            onlinePaymentService.updatePaymentStatus(payment.getPaymentId(), new PaymentReturnDto(true, false, false, false), card);
+                        }
+                        onlinePaymentService.updatePaymentStatus(payment.getPaymentId(), paymentReturnDto, card);
+                        if (transactionId != null) {
+                            onlinePaymentService.setPaytrailTransactionId(payment.getPaymentId(), transactionId);
+                            paymentPaytrailService.sendMitChargeNotify(payment.getOrderId());
                         }
                     } catch (Exception e) {
+                        log.debug("subscription renewal payment failed for orderId: " + payment.getOrderId(), e);
                         sendNotificationService.sendErrorNotification(
```

Comment:
Tässä tulis sitten meille error notifikaatti kaikista virheistä mitä tuolla tapahtuu mutta pitäiskö tässäkin heittää lopuksi virhe vielä ylöspäin että nuo lopun virheenkäsittelyt sen hanskais?